pkgs/dart_mcp_server/lib/src/mixins/dtd.dart

```diff
@@ -997,16 +1031,21 @@ base mixin DartToolingDaemonSupport
     inputSchema: Schema.object(),
   );
 
-  static final _connectedAppsNotSupported = CallToolResult(
-    isError: true,
-    content: [
-      TextContent(
-        text:
-            'A Dart SDK of version 3.9.0-163.0.dev or greater is required to '
-            'connect to Dart and Flutter applications.',
-      ),
-    ],
-  )..failureReason = CallToolFailureReason.connectedAppServiceNotSupported;
+  CallToolResult get _connectedAppsNotSupported {
+    String errorText;
+    if (_serviceDetectionError != null) {
+      errorText = 'ConnectedApp service not available: $_serviceDetectionError';
```

Comment:
Note that in this case it is usually going to be because the user isn't on the required SDK version, so it would be good to include that in the text here?

Reply:
@lukemmtt it would be good to resolve this comment as well if you can :)

Reply:
Hey @jakemac53 , no problem, although I'm not certain I understood. Is this what you had in mind?
Commit: https://github.com/dart-lang/ai/pull/257/commits/a91675f89234f69b147b17b12466d5978e8b56e8